framework/src/interfaces/TaggingInterface.C

```diff
@@ -224,13 +269,19 @@ TaggingInterface::accumulateTaggedLocalResidual()
 {
   for (auto & re : _re_blocks)
     *re += _local_re;
+  for (auto & ref : _absref_blocks)
+    for (const auto i : index_range(_local_re))
+      (*ref)(i) += std::abs(_local_re(i) + std::numeric_limits<Number>::min());
```

Comment:
Why is the `min()` addition necessary?

Reply:
This is to prevent a divide-by-zero in ReferenceResidualProblem when taking the local norm.

Reply:
I would add the `min()` after the absolute value then, in the perhaps infinitely small chance that `_local_re(i)` is equal to `-std::numeric_limits<Number>::min()`

Reply:
Actually I can just do this once huh...

Reply:
This is now done in `ReferenceResidualProblem`. `TaggingInterface` is kept clean with no connection to "reference" values.